### DISCUSSION TITLE
[Porousflow] PorousFlowSink is independent to time for some reason

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#22645] [Q&A Modules: General] [answered]
Hi guys,

so my `PorousFlowSink` is independent to time for some reason
```
[rainfall_recharge]
   type = PorousFlowSink
    boundary = 'left'
    variable = porepressure
    flux_function = -1
     save_in = fluxes_out
[]
```

I check the total flux via `NodalSum`

```
  [left_flux]
    type = NodalSum
    boundary = 'left'
    variable = fluxes_out
    execute_on = 'timestep_end'
  []
```

here are my [Executioner] and [PorousFlowUnsaturated] blocks:

```
[PorousFlowUnsaturated]
porepressure = porepressure
coupling_type = Hydro
gravity = '0 0 -9.81'
fp = the_simple_fluid
time_unit = years
add_darcy_aux = false
[]
```

```
[Executioner]
type = Transient
start_time = 0
dt = 0.001
end_time = 0.01
solve_type = NEWTON
petsc_options = '-snes_converged_reason'
petsc_options_iname = '-pc_type -pc_hypre_type'
petsc_options_value = 'hypre    boomeramg'
nl_rel_tol = 1e-3
nl_abs_tol = 8e0
l_tol = 1e-5
l_max_its = 150
nl_max_its = 50
line_search = none
automatic_scaling = true
[]
```

the surface area of the 'left' boundary is 350 m2
from the document flux_function of the PorousFlowSink is in kg.m-2.s-1
As can be seen above, my time unit is 'year' and dt is 0.001

HOWEVER, the Nodalsum still gives me 350 kg which is the case for time unit = sec and dt = 1

and doesn't matter what the dt is, it still gives me the same value

Postprocessor Values:
+----------------+----------------+
| time           | left_flux      |
+----------------+----------------+
|   0.000000e+00 |   0.000000e+00 |
|   1.000000e-03 |  -3.500000e+02 |
|   2.000000e-03 |  -3.500000e+02 |
|   3.000000e-03 |  -3.500000e+02 |
|   4.000000e-03 |  -3.500000e+02 |
|   5.000000e-03 |  -3.500000e+02 |
|   6.000000e-03 |  -3.500000e+02 |
|   7.000000e-03 |  -3.500000e+02 |
|   8.000000e-03 |  -3.500000e+02 |
|   9.000000e-03 |  -3.500000e+02 |
|   1.000000e-02 |  -3.500000e+02 |
+----------------+----------------+

Not sure if I've missed anything, if you have an idea please let me know.

Thank you!
Traiwit

**Comment by Traiwit** [comment]
The same thing also happens when I checked with

/modules/porous_flow/examples/groundwater/ex02_steady_state.i

Not sure if NodalSum does not read the flux properly or if something is wrong with the BC

**Comment by WilkAndy** [comment, marked as answer]
Hi @Traiwit ,

Have a look at https://mooseframework.inl.gov/modules/porous_flow/boundaries.html .   You'll see the `NodalSum` is measured in kg/s.   In your case, it will be kg/year - see the "Alternative time units" section of https://mooseframework.inl.gov/source/materials/PorousFlowSingleComponentFluid.html , viz "fluid sources must be measured in kg/m3/time" (and areal sources in kg/m2/time , but i didn't write that in the doco, ooops).

**Reply by Traiwit** [reply]
Noted, thanks @WilkAndy 

I thought `NodalSum` would sum to mass (or volume) that passes thru the boundary at a particular time.

just wondering is there any object that can do that kinda thing?

**Reply by WilkAndy** [reply]
Not that i know of, @traiwit .   I usually use a Postprocessor to capture "dt", and then a ParsedAux to multiply by it in order to get the mass flowing through the boundary for a particular time step
